addon/modifiers/in-viewport.js

```diff
@@ -0,0 +1,114 @@
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { DEBUG } from '@glimmer/env';
+import Modifier from 'ember-class-based-modifier';
+import deepEqual from 'fast-deep-equal';
+
+const WATCHED_ELEMENTS = DEBUG ? new WeakSet() : undefined;
+
+export default class InViewportModifier extends Modifier {
+  @service inViewport;
+
+  name = 'in-viewport';
+
+  lastOptions;
+
+  get options() {
+    // eslint-disable-next-line no-unused-vars
+    const { onEnter, onExit, ...options } = this.args.named;
+    return options;
+  }
+
+  get hasStaleOptions() {
+    !deepEqual(this.options, this.lastOptions);
```

Comment:
Whoops, missed a `return` here 😄 
```suggestion
    return !deepEqual(this.options, this.lastOptions);
```